src/test/java/com/bravo/user/service/AddressServiceTest.java

```diff
@@ -0,0 +1,47 @@
+package com.bravo.user.service;
+
+import com.bravo.user.dao.model.Address;
+import com.bravo.user.dao.model.mapper.ResourceMapper;
+import com.bravo.user.dao.repository.AddressRepository;
+import com.bravo.user.model.dto.AddressDto;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@ExtendWith(MockitoExtension.class)
+class AddressServiceTest {
+    private AddressService testObj;
+    @Mock
+    private AddressRepository repository;
+    @Autowired
+    private ResourceMapper mapper;
```

Comment:
Sometimes a field gets mocked but don't get called in a `when()`, then intellij says that variable is unused, which is not true but it messes with you when you're reading it.

With the mapper I think it being a utility component it's fine to have additional coverage on it, also gets rid of the need to do another `when()`

I call all subjectUnderTest `testObj` so that way it's quick and easy to see what method is the test about. I usually middle click on it to bring the cursor to the type, or by looking at the test file name or the class name.

https://stackoverflow.com/questions/7321407/what-is-sut-and-where-did-it-come-from

Keeping the test object name consistent makes it easy to spot in any test classes. I have all my tests' SUT named testObj, and the reason for having all tests using the same test object name is because all the tests always have the same purpose of running the SUT method and check results. The business logic varies, but tests' purpose and flow stay the same.